assets/src/edit-story/components/header/buttons/downloadZip.js

```diff
@@ -67,64 +67,67 @@ function DownloadZip() {
-      image: [],
-    };
-
-    currentPage.elements.forEach((element) => {
-      if (element?.type in elements) {
-        elements[element.type].push(element);
-      }
-    });
-
     const zip = new JSZip();
 
+    const mediaTypes = ['image', 'video'];
     await Promise.all(
-      Object.keys(elements).map(async (mediaType) => {
-        const mediaElements = elements[mediaType];
-
-        if (mediaElements.length) {
-          await Promise.all(
-            mediaElements.map(async (mediaElement, index) => {
-              const { src, mimeType, id } = mediaElement.resource;
-              const extension = COMMON_MIME_TYPE_MAPPING[mimeType];
-
-              if (!extension) {
-                return;
-              }
-
-              let imageSrc = src;
-
-              /**
-               * Download double size image of the page ratio as unsplash image size can be very
-               * large and we aren't using srcset to keep things simple for the user.
-               */
-              if (src.startsWith('https://images.unsplash.com')) {
-                imageSrc = addQueryArgs(src, {
-                  w: PAGE_WIDTH * 2,
-                  h: (PAGE_WIDTH * 2) / PAGE_RATIO,
-                });
-              }
-
-              const resp = await fetch(imageSrc);
-              const respBlob = await resp.blob();
-              const fileName = `${mediaType}-${index + 1}.${extension}`;
-              const file = new File([respBlob], fileName);
-
-              const elementIndex = currentPage.elements.findIndex(
-                (element) => element?.resource?.id === id
-              );
-
-              if (elementIndex) {
-                currentPage.elements[elementIndex].resource.src = fileName;
-              }
-
-              const encodedUrl = src.replaceAll('&', '&amp;'); // To match url in the rendered markup.
-              markup = markup.replace(encodedUrl, fileName);
-
-              zip.file(fileName, file);
-            })
-          );
+      currentPage.elements.map(async (element) => {
+        const mediaType = element.type;
+
+        if (!mediaTypes.includes(mediaType)) {
+          return;
+        }
+        const { id } = element;
+        const { src, mimeType, poster } = element.resource;
+        const extension = COMMON_MIME_TYPE_MAPPING[mimeType];
+
+        if (!extension) {
+          return;
+        }
+
+        let imageSrc = src;
+
+        /**
+         * Download double size image of the page ratio as unsplash image size can be very
+         * large and we aren't using srcset to keep things simple for the user.
+         */
+        if (src.startsWith('https://images.unsplash.com')) {
+          imageSrc = addQueryArgs(src, {
+            w: PAGE_WIDTH * 2,
+            h: (PAGE_WIDTH * 2) / PAGE_RATIO,
+          });
+        }
+
+        const resp = await fetch(imageSrc);
+        const respBlob = await resp.blob();
+        const fileName = `${mediaType}-${id}.${extension}`;
```

Comment:
This will create image names as `image-1e46ca05-23e2-48e9-95c9-fd90f2113405.jpeg`. IMO using id will confuse the user when uploading multiple downloaded images to ad manager. I think an index-based name is easier to work with.

Reply:
Using index based name now.